iml-agent/src/network_interfaces.rs

```diff
@@ -38,3 +46,51 @@ pub async fn get_interfaces() -> Result<Vec<NetworkInterface>, ImlAgentError> {
 
     parse_interfaces(network_interfaces, net_stats)
 }
+
+pub async fn get_lnet_data() -> Result<LNet, ImlAgentError> {
+    let output = get_lnet_data_cmd().checked_output().await?;
+
+    let lnet_data = std::str::from_utf8(&output.stdout)?;
+
+    let xs: LNet = serde_yaml::from_str(lnet_data)?;
+
+    Ok(xs)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_lnetctl_net_show_output() {
+        let data = r#"net:
+    - net type: lo
+      local NI(s):
+        - nid: 0@lo
+          status: up
+    - net type: tcp
+      local NI(s):
+        - nid: 10.73.20.21@tcp
+          status: up
```

Comment:
nit: could have one or two with `status: down`

Reply:
Sounds good, i'll add that.